baseplate/secrets/fetcher.py

```diff
@@ -108,16 +111,73 @@ def ttl_to_time(ttl):
 
 class VaultClientFactory(object):
     """Factory that makes authenticated clients."""
-    def __init__(self, base_url, role):
+    def __init__(self, base_url, role, auth_type, auth_path):
```

Comment:
If I understand correctly, this may break backward compatibility with things in the wild currently using VaultClientFactory. If that is the case, we should probably provide kwarg defaults to preserve the current EC2 auth default.

Reply:
Couldn't find any reference to things using this, so we may be safe with just passing in the defaults through the configparser. No strong feelings towards either method though.

Reply:
Sounds like a @spladug question, then. I will defer to his wisdom.

Reply:
It's just an internal helper for this file. Should be safe.